django_cte/meta.py

```diff
@@ -72,24 +72,29 @@ def relabeled_clone(self, relabels):
 
 class CTEColumnRef(Expression):
 
-    def __init__(self, name, output_field):
+    def __init__(self, name, cte_name, output_field):
         self.name = name
+        self.cte_name = cte_name
         self.output_field = output_field
         self._alias = None
 
-    def resolve_expression(self, query=None, allow_joins=True, reuse=None,
-                           summarize=False, for_save=False):
-        ref = self.copy()
-        ref._alias = query.get_initial_alias()
-        return ref
-
     def relabeled_clone(self, relabels):
-        if self._alias is not None and self._alias in relabels:
-            clone = self.copy()
+        if (
+            self.cte_name not in relabels
+            and self._alias not in relabels
```

Comment:
If I understood what happened correctly, `relabeled_clone` gets called when the query is used as a subquery. So if the wrapping query were then used as another subquery might trigger this again? It should be easy to check when I dive back in.

To answer your question, I don't know. My guess is that it could be a key in `relabels` simply because this method is based off of the `Col` expression's implementation.